pymeasure/instruments/keithley/keithley6517b.py

```diff
@@ -0,0 +1,343 @@
+        :param voltage: Upper limit of voltage in Volts, from -1000 V to 1000 V
+        :param auto_range: Enables auto_range if True, else uses the
+                           voltage_range attribut
+        """
+        log.info("%s is measuring voltage.", self.name)
+        self.write(":SENS:FUNC 'VOLT';"
+                   ":SENS:VOLT:NPLC %f;" % nplc)
+        if auto_range:
+            self.write(":SENS:VOLT:RANG:AUTO 1;")
+        else:
+            self.voltage_range = voltage
+        self.check_errors()
+
+    def measure_current(self, nplc=1, current=1.05e-4, auto_range=True):
+        """ Configures the measurement of current.
+
+        :param nplc: Number of power line cycles (NPLC) from 0.01 to 10
+        :param current: Upper limit of current in Amps, from -21 mA to 21 mA
+        :param auto_range: Enables auto_range if True, else uses the
+                           current_range attribut
+        """
+        log.info("%s is measuring current.", self.name)
+        self.write(":SENS:FUNC 'CURR';"
+                   ":SENS:CURR:NPLC %f;" % nplc)
+        if auto_range:
+            self.write(":SENS:CURR:RANG:AUTO 1;")
+        else:
+            self.current_range = current
+        self.check_errors()
+
+    def auto_range_source(self):
+        """ Configures the source to use an automatic range.
+        """
+        self.write(":SOUR:VOLT:RANG:AUTO 1")
+
+    def apply_voltage(self, voltage_range=None,
+                      compliance_current=1e-5):
+        """ Configures the instrument to apply a source voltage, and
+        uses an auto range unless a voltage range is specified.
+        The compliance current is also set.
+
+        :param compliance_current: A float in the correct range for a
+                                   :attr:`~.Keithley6517B.compliance_current`
+        :param voltage_range: A :attr:`~.Keithley6517B.voltage_range` value
+                              or None (activates auto range)
+        """
+        log.info("%s is sourcing voltage.", self.name)
+        self.source_mode = 'voltage'
+        if voltage_range is None:
+            self.auto_range_source()
+        else:
+            self.source_voltage_range = voltage_range
+        self.compliance_current = compliance_current
+        self.check_errors()
+
+    @property
+    def error(self):
+        """ Returns a tuple of an error code and message from a
+        single error. """
+        err = self.values(":system:error?")
+        if len(err) < 2:
+            err = self.read()  # Try reading again
+        code = err[0]
+        message = err[1].replace('"', '')
+        return (code, message)
+
+    def check_errors(self):
+        """ Logs any system errors reported by the instrument.
+        """
+        code, message = self.error
+        while code != 0:
+            t = time.time()
+            log.info("Keithley 2450 reported error: %d, %s", code, message)
```

Comment:
Overlooked old instrument name. 
If we factor that out, we could use `self.__class__.__name__` or somesuch to make this dynamically correct.